core/input_types.ts

```diff
@@ -19,5 +19,7 @@ export enum inputTypes {
   // A down-facing block stack.  E.g. 'if-do' or 'else'.
   STATEMENT = ConnectionType.NEXT_STATEMENT,
   // A dummy input.  Used to add field(s) with no input.
-  DUMMY = 5
+  DUMMY = 5,
+  // An unknown type of input defined by an external developer.
+  CUSTOM = 5
```

Comment:
Maybe add a comment explaining why DUMMY and CUSTOM are both 5.
Also, are trailing commas are legal with TypeScript enums.  Isn't that desirable?

Reply:
Oh whoops actually meant to make it 6, thanks for the catch!